libraries/AP_HAL/AP_HAL_Boards.h

```diff
@@ -188,6 +188,10 @@
 #define AP_EXTENDED_DSHOT_TELEM_V2_ENABLED HAL_REQUIRES_BDSHOT_SUPPORT
 #endif
 
+#ifndef AP_EXTENDED_ESC_TELEM_ENABLE
```

Comment:
This needs to be a different name - far too generic. 
```
AP_EXTENDED_APD_ESC_TELEM_ENABLED
```
would be ok with me

Reply:
This is not specific to APD (there was an APD-specific extension to BLHeli in the PR previously, but that has been split out). This PR is just for handling the [uavcan.equipment.esc.StatusExtended](https://github.com/dronecan/DSDL/blob/master/uavcan/equipment/esc/1036.StatusExtended.uavcan) message